src/parkapi_sources/converters/apcoa/mapper.py

```diff
@@ -0,0 +1,101 @@
+    @staticmethod
+    def map_static_parking_site(apcoa_input: ApcoaParkingSiteInput) -> StaticParkingSiteInput:
+        latitude, longitude = next(
+            iter(
+                (navigation_locations_input.GeoCoordinates.Latitude, navigation_locations_input.GeoCoordinates.Longitude)
+                for navigation_locations_input in apcoa_input.NavigationLocations
+                if navigation_locations_input.LocationType == ApcoaNavigationLocationType.DEFAULT
+            )
+        )
+
+        static_parking_site_input = StaticParkingSiteInput(
+            uid=str(apcoa_input.CarParkId),
+            name=apcoa_input.CarparkLongName if apcoa_input.CarparkLongName else apcoa_input.CarparkShortName,
+            lat=latitude,
+            lon=longitude,
+            purpose=PurposeType.CAR,
+            type=apcoa_input.CarparkType.Name.to_parking_site_type_input(),
+            has_realtime_data=False,  # TODO: change this as soon as Apcoa API offers realtime data
+            public_url=apcoa_input.CarParkWebsiteURL,
+            static_data_updated_at=apcoa_input.LastModifiedDateTime,
+        )
+
+        if apcoa_input.Address.Street and apcoa_input.Address.Zip and apcoa_input.Address.City:
+            static_parking_site_input.address = f'{apcoa_input.Address.Street}, {apcoa_input.Address.Zip} {apcoa_input.Address.City}'
+
+        if apcoa_input.CarParkPhotoURLs:
+            static_parking_site_input.photo_url = apcoa_input.CarParkPhotoURLs.CarparkPhotoURL1
+
+        apcoa_weekdays = [
+            ApcoaOpeningHoursWeekdays.MONDAY,
+            ApcoaOpeningHoursWeekdays.TUESDAY,
+            ApcoaOpeningHoursWeekdays.WEDNESDAY,
+            ApcoaOpeningHoursWeekdays.THURSDAY,
+            ApcoaOpeningHoursWeekdays.FRIDAY,
+            ApcoaOpeningHoursWeekdays.SATURDAY,
+            ApcoaOpeningHoursWeekdays.SUNDAY,
+        ]
+        apcoa_opening_times = defaultdict(list)
+        for opening_hours_input in apcoa_input.OpeningHours:
+            opening_time = opening_hours_input.OpeningTimes if opening_hours_input.OpeningTimes != ApcoaOpeningHoursTime.CLOSED else 'off'
+            if opening_hours_input.Weekday in apcoa_weekdays[:-2]:
+                apcoa_opening_times[opening_time].append(opening_hours_input.Weekday.to_osm_opening_day_format())
+        for opening_time, weekdays in apcoa_opening_times.items():
+            if len(weekdays) == 7 and opening_time == apcoa_opening_times[ApcoaOpeningHoursTime.OPEN_24H]:
+                static_parking_site_input.opening_hours = '24/7'
+            elif (
+                len(weekdays) == 5
+                and ApcoaOpeningHoursWeekdays.SATURDAY not in weekdays
+                and ApcoaOpeningHoursWeekdays.SUNDAY not in weekdays
+            ):
+                static_parking_site_input.opening_hours = f'Mo-Fr {opening_time}'
+            else:
+                static_parking_site_input.opening_hours = ';'.join([f'{weekday} {opening_time}' for weekday in weekdays])
+
```

Comment:
After I really need some concentration to understand what happens in OSM transformation: this would be a good part of code to move to a method just for that purpose, because it's far from trivial, and it would make sense to make a unittest with parameters to test different inputs. At our validators, we have some examples for tests of multiple inputs: https://github.com/ParkenDD/parkapi-sources-v3/blob/main/tests/validators/integer_validators_test.py#L14

Reply:
It is now made a method for the purpose. Is it okay in the validator class?

Reply:
Yes, the approach is good, but this is really something which makes sense to create unittests, as there are a lot of different inputs.